### DISCUSSION TITLE
What is the best way to load gene annotations and make them searchable?

### DISCUSSION THREAD

**Discussion by jpcartailler** [GMOD/jbrowse-components#3875] [Q&A] [answered]
Greetings,

I cannot get searchable genes to work in JBrowse2 and am seeking recommendations. Here is what I've tried (in @jbrowse/cli/2.6.3 win32-x64 node-v16.16.0)

**OPTION 1**

Loaded in a bed file with simple gene model:
> ![image](https://github.com/GMOD/jbrowse-components/assets/5258731/0350411e-5959-426a-920a-a73a9315d937)
* I can see it
* I cannot index it (`jbrowse text-index --file=genes.bed` yields ` Error: Unsupported file type genes.bed`)

**OPTION 2**

Tried to load `gencode.vM25.basic.annotation.gtf.gz` from Gencode (which unpacks to ~629MB. The file loads, but the track shows the following error:
> ![image](https://github.com/GMOD/jbrowse-components/assets/5258731/96d08f91-82d2-4202-850c-1548a020f754)

Is there a best practice or working solution for getting in gene annotations and being able to search them?  Appreciate any guidance or leads. Cheers!

**Comment by cmdcolin** [comment, marked as answer]
the best way is to use tabix indexed gff. we do not support text indexing from either gtf or bed files currently.

one "easy-ish" way to convert gtf to gff is 

```
sudo apt install gffread # for converting gtf to gff
sudo apt install genometools # for sorting gff
gffread -E file.gtf -o- > file.gff3
gt gff3 -sortlines -tidy -retainids file.gff > file.sorted.gff
bgzip file.sorted.gff
tabix file.sorted.gff.gz
jbrowse add-track file.sorted.gff.gz --load copy --out /path/to/your/instance
jbrowse text-index --out /path/to/your/instance # indexes all tracks that it recognizes in the /path/to/your/instance/config.json
```

after those steps, your instance should be text searchable using the little 'location box', you can enter gene names there

see quickstart here for these steps outlined https://jbrowse.org/jb2/docs/quickstart_web/#adding-a-gff3-file-with-gff3tabix


your post did make me reconsider the idea of loading BED and GTF better in JBrowse, and also we can work on removing that 512Mb limit...I had fixed removing that limit in another file type. it is somewhat of a warning to the user because it is trying to unpack a large amount of data in memory, but we should allow it still. the tabix indexed gff allows the program to only unpack small amounts of data at a time in memory which is nice, but it is a tricky process to prepare the tabix indexed gffs.
